### PR TITLE
Make http_client compile in C++11 and above

### PR DESCRIPTION
**Pull request by Morosko**

### CODE REVIEW

File: dlib/http_client/http_client.cpp [hunk 1]
```diff
@@ -454,9 +454,11 @@ namespace dlib
                 for (stringmap::const_iterator si = ci->second.begin(); si != ci->second.end(); ++si)
                 {
                     std::ifstream in(si->second.c_str());
+					std::string file_content((std::istreambuf_iterator<char>(in)),
+                                    std::istreambuf_iterator<char>());
                     postBody << "--" << mime_boundary << "\r\n"
                         "Content-Disposition: form-data; name=\"" << ci->first << "\"; filename=\"" << get_basename(si->second) << "\"\r\n\r\n"
-                        << in << "\r\n";
```
Comment (davisking) [inline dlib/http_client/http_client.cpp]: Doesn't putting in.rdbuf() here work?  That should be a more efficient solution.  Also, please fix the messed up tabbing.

**Comment by davisking** [comment]
Cool, thanks for the PR :)

**Comment by Morosko** [comment]
Thanks for the fast reply and the hint to make it more efficient! I changed it now.

**Comment by davisking** [comment]
No problem. Thanks for the PR :)

**Pull request merged**